src/lib.rs

```diff
@@ -237,16 +238,17 @@
 #![forbid(missing_docs, unsafe_op_in_unsafe_fn)]
 #![cfg_attr(not(feature = "std"), no_std)]
 #![cfg_attr(feature = "alloc", feature(allocator_api))]
-#![cfg_attr(feature = "alloc", feature(new_uninit))]
-#![cfg_attr(feature = "alloc", feature(get_mut_unchecked))]
+#![cfg_attr(feature = "arc", feature(get_mut_unchecked))]
```

Comment:
```suggestion
#![cfg_attr(
    all(any(feature = "std", feature = "alloc"), feature = "arc"),
    feature(get_mut_unchecked)
)]
```

Reply:
hmm, but now one can activate the `have_allocation` and the `arc` feature without activating `std` and `alloc`, right? I also don't like the constant repeating of the bound, but I think the crate should compile with any combination of features.

Reply:
Wait I know what to do :) It needs a tiny build.rs but it's the best of both worlds.